### PR TITLE
[sharded][3] use state value with key hash

### PR DESCRIPTION
**Pull request by areshand**
## Description
StateValueIndex is removed from writing in this PR. 

backup and restore shouldn't be impacted 
https://github.com/aptos-labs/aptos-core/blob/populate_new_state_kv_schema/storage/aptosdb/src/state_store/mod.rs#L1049 
backup uses jmt get keys and then use state key to state store to get state value 

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [x] Validator Node
- [x] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Existing test and spot check generated DBs

replay-verify with sharding mode on and no error except timeout
https://github.com/aptos-labs/aptos-core/actions/runs/9585888369

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by grao1991 — APPROVED** [review]

File: storage/aptosdb/src/pruner/state_kv_pruner/state_kv_metadata_pruner.rs [hunk 1]
```diff
@@ -39,14 +39,13 @@ impl StateKvMetadataPruner {
                 let mut iter = self
                     .state_kv_db
                     .db_shard(shard_id)
-                    .iter::<StaleStateValueIndexSchema>()?;
+                    .iter::<StaleStateValueIndexByKeyHashSchema>()?;
```
Comment (msmouse) [inline storage/aptosdb/src/pruner/state_kv_pruner/state_kv_metadata_pruner.rs:42]: Need to keep old pruner working?

File: storage/aptosdb/src/pruner/state_kv_pruner/state_kv_shard_pruner.rs [hunk 2]
```diff
@@ -50,15 +51,17 @@ impl StateKvShardPruner {
     ) -> Result<()> {
         let batch = SchemaBatch::new();
 
-        let mut iter = self.db_shard.iter::<StaleStateValueIndexSchema>()?;
+        let mut iter = self
+            .db_shard
+            .iter::<StaleStateValueIndexByKeyHashSchema>()?;
```
Comment (msmouse) [inline storage/aptosdb/src/pruner/state_kv_pruner/state_kv_shard_pruner.rs:56]: ditto

File: storage/aptosdb/src/rocksdb_property_reporter.rs [hunk 3]
```diff
@@ -139,7 +139,7 @@ fn update_rocksdb_properties(
             set_property(cf, ledger_db.transaction_accumulator_db_raw())?;
         }
 
-        for cf in state_kv_db_column_families() {
+        for cf in state_kv_db_new_key_column_families() {
```
Comment (msmouse) [inline storage/aptosdb/src/rocksdb_property_reporter.rs:142]: We just break the dashboard for the old format?

File: storage/aptosdb/src/schema/mod.rs [hunk 4]
```diff
@@ -105,10 +105,16 @@ pub mod fuzzing {
...
+            assert_no_panic_decoding::<
+                super::stale_state_value_index_by_key_hash::StaleStateValueIndexByKeyHashSchema,
+            >(data);
             assert_no_panic_decoding::<super::stale_state_value_index::StaleStateValueIndexSchema>(
                 data,
             );
             assert_no_panic_decoding::<super::state_value::StateValueSchema>(data);
+            assert_no_panic_decoding::<super::state_value_by_key_hash::StateValueByKeyHashSchema>(
+                data,
+            );
```
Comment (msmouse) [inline storage/aptosdb/src/schema/mod.rs:117]: 👍

File: storage/aptosdb/src/state_store/mod.rs [hunk 5]
```diff
@@ -269,6 +274,37 @@ impl StateDb {
                 })
             })
     }
+
+    fn get_state_value_with_version_by_version(
```
Comment (msmouse) [inline storage/aptosdb/src/state_store/mod.rs:278]: you intended to use this somewhere?

File: storage/aptosdb/src/pruner/state_kv_pruner/state_kv_metadata_pruner.rs (hunk 1 above)
Comment (areshand) [reply storage/aptosdb/src/pruner/state_kv_pruner/state_kv_metadata_pruner.rs:42]: This schema should be removed before sharding migration. so safe to delete?

File: storage/aptosdb/src/rocksdb_property_reporter.rs (hunk 3 above)
Comment (areshand) [reply storage/aptosdb/src/rocksdb_property_reporter.rs:142]: updated. Probably need to update dashboard to show different CFs when in migration

File: storage/aptosdb/src/state_store/mod.rs (hunk 5 above)
Comment (areshand) [reply storage/aptosdb/src/state_store/mod.rs]: deleted

File: storage/aptosdb/src/pruner/state_kv_pruner/state_kv_metadata_pruner.rs (hunk 1 above)
Comment (msmouse) [reply storage/aptosdb/src/pruner/state_kv_pruner/state_kv_metadata_pruner.rs:42]: no? your binary needs to run the old db same as before?

Comment (msmouse) [reply storage/aptosdb/src/pruner/state_kv_pruner/state_kv_metadata_pruner.rs:42]: I see.

**Review by msmouse — APPROVED** [review]

**Pull request merged**